src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        int figureCount = 5;
+        Random randResult = new Random();
+        int figureNum = randResult.nextInt(figureCount);
+
+        double randomRange = 100.0;
+
+        ColorSupplier randomColor = new ColorSupplier();
+
+        String color = randomColor.getRandomColor();
+
+        switch (figureNum) {
+            case 0:
+                double randomSide = new Random().nextDouble() * randomRange;
```

Comment:
```suggestion
                double randomSide = randResult.nextDouble() * randomRange;
```
Коли ти звертаєшся до ключового слова `new` це означає, що викличеться конструктор і створиться новий об'єкт, тому оптимальніше створити один об'єкт на метод і його всюди використати замість того щоб створювати при кожному виклику методу 10+ об'єктів, це може бути затратно